docs/nsm_api.md

```diff
@@ -0,0 +1,445 @@
+        <ul>
+            <li>Connectivity to Isolated Resources</li>
+            <li>Protection for Threats</li>
+            <li>Guaranteed Bandwidth/Latency</li>
+            <li>Load Balancing</li>
+            <li>Proxying</li>
+        </ul>
+        A Network Service has a name and a payload.  The payload defines the type of L2/L3 payload (Ethernet/IPv4/IPv6/MPLS/etc) the Network Service accepts.
+        Examples:
+        <ul>
+            <li>Secure Intranet Access</li>
+            <li>Bride Domains</li>
+            <li>Routing Domains (VRFs)</li>
+            <li>Cloud-native Network Functions (CNFs) of all sorts</li>
+        </ul>
+    </dd>
+    <dt>Network Service Client (NSC)</dt>
+    <dd>
+        A Network Service Client is an entity which wishes to connect to a Network Service.  
+        Example: A Pod in Kubernetes which wants to connect to a Network Service.
+    </dd>
+    <dt>L2/L3 Connection</dt>
+    <dd>
+        An L2/L3 Connection is simply some mechanism which can carry L2/L3 traffic (IP/Ethernet/MPLS/etc) bidirectionally between the Network Service Client and the Network Service Endpoint.  L2/L3 Connections are simple cross connects.  They are not bridge domains.  A bridge domain is itself a Network Service, and can be connected to via an L2/L3 Connection.
+    </dd>
+    <dt>Network Service Endpoint (NSE)</dt>
+    <dd>
+        A Network Service Endpoint provides a concrete usable instance of a Network Service for consumption by Network Service Clients.  Example: Pods providing a Network Service in Kubernetes.
+    </dd>
+</dl>
+
+![Network Service Mesh Abstract Components](./images/NSM%20Domains.png)
+<dl>
+    <dt>Network Service Dataplane (NSD)</dt>
+    <dd>
+        Within the Network Service Manager domain, the Network Service Dataplan is the dataplane managed by the Network Service Manager to 
+        <ul>
+            <li>Connect Network Service Clients in its domain to Network Service Endpoints.  The Network Service Endpoints need not be in the Network Service Managers domain.</li>
+            <li>Connect Network Service Endpoints in its domain to Network Service Clients.  The Network Service Clients need not be in the Network Service Managers domain, nor within any Service Registry Domain that the Network Service Manager participates in.</li>
+        <ul>
+    </dd>
+    <dt>Network Service Manager (NSM)</dt>
+    <dd>
+        A Network Service Manager (NSM) manages a colletion of Network Service Clients, Network Service Endpoints, and the Network Service Dataplane for those NSCs and NSEs within its Network Service Manager Domain.  An example of a Network Service Manager Domain would be a single Kubernetes Node.  A Network Service Manager is responsible to
+        <ul> 
+            <li>Advertise Network Service Endpoints in its domain to zero or more Network Service Registries</li>
```

Comment:
Could you clarify what you mean by `more Network Service Registries`? If there are 2 k8s clusters, it seems you expect NSM to advertise NSE to both?

Reply:
@sbezverk I don't anticipate an NSM on a Node advertising to more than one Network Service Registries, just the one for the K8s Cluster.  I do anticipate a Network Service Manager for something not-K8s-clustery possibly advertising to multiple K8s Clusters (and their Network Service Registries).

Reply:
hmm, making NSM topology aware will make it more complex and in addition it would take on a role which is not its primary business. I believe if inter cluster  exchange is required, it should be done either by a multi cluster offered infra or a completely separate controller.